### ISSUE TITLE
SMIT default example should be GPU-poor friendly

### ISSUE THREAD

**Issue by Thytu** [Thytu/SMIT#12]
Once #8 is merged the default example will consume a considerable amount of VRAM (~77Go) which prevents many potential users to test SMIT.

SMIT has been designed to be GPU-poor friendly from the beginning and its default example should showcase it.

There is still plenty of room for improvement to reduce VRAM usage, some of are:
* Changing the optimizer : `galore_adamw` and `adamw_bnb_8bit` could be a good options
* LoRA: it has been deactivated when fixing #8 but a solution to re-enable it should be investigated

For more ideas: [Methods and tools for efficient training on a single GPU](https://huggingface.co/docs/transformers/perf_train_gpu_one)

**Comment by Thytu** [comment]
More information on optimizer's impact over VRAM usage

![image](https://github.com/Thytu/SMIT/assets/43698357/1432085c-3b20-4ed1-a7eb-660087cc99fd)

**Comment by Thytu** [comment]
Batch size of 2 with gradient accumulation of 4 works but takes ~400mins (~6h) to converge which is way too long. Trying to increase BS by 1

<img width="591" alt="image" src="https://github.com/Thytu/SMIT/assets/43698357/2adefc42-0656-4c0f-9170-dc1b28172db1">

<img width="577" alt="image" src="https://github.com/Thytu/SMIT/assets/43698357/8d7d77da-81f3-4d55-bd46-749452777bfc">

**Comment by Thytu** [comment]
Increasing BS to 3 resulting in a VRAM above 40GB (~46GB). I'll probably release it with Quant nf4 BS2 GA4.
